core/src/main/java/org/apache/accumulo/core/iterators/Combiner.java

```diff
@@ -149,6 +160,38 @@ public void next() throws IOException {
 
   private Key workKey = new Key();
 
+  private static final Cache<String,Long> loggedMsgCache = CacheBuilder.newBuilder().expireAfterWrite(1, TimeUnit.HOURS).maximumSize(10000).build();
+
+  private void sawDelete() {
+    if (isPartialCompaction) {
+      switch (deleteHandlingAction) {
+        case LOG_ERROR:
+          try {
+            loggedMsgCache.get(this.getClass().getName(), new Callable<Long>() {
```

Comment:
At first, I thought "Callable<Void>" might be better, but apparently CacheBuilder doesn't like that, so might be better to use "Callable<Boolean>", because it's more readable to see "Boolean.TRUE" inserted into the cache to denote that a log message has recently occurred. (Unless you were going to use the integer to denote how many times it has been logged before suppressing repeats after some threshold, like 5.)